pkg/api/aim/runs.go

```diff
@@ -171,23 +171,20 @@ func GetRunMetrics(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
 	}
 
-	metricKeysMap, contexts := make(fiber.Map, len(b)), make([]types.JSONB, 0, len(b))
+	metricKeysMap := make(map[string]map[string]types.JSONB, len(b))
```

Comment:
This has become a really weird structure, especially with this new GetHash function. We receive a list of key ("name") and context tuples from the client, we can just use that as the key and `nil` as the value so we have a proper unordered unique list of key and context tuples.

We can then build the subquery with these tuples.

Finally we can create another map of these tuples in which we store a list of iters and a list of values **and nothing else**. Duplicating the key/name and context in there does not make sense! Once this map is built, we can return the proper structure to the client.

Reply:
could we please use actual structs instead of concatenated strings? merging and splitting with underscores could introduce all kinds of bugs!